### ISSUE TITLE
Fix error 'metadata_shape' not used when training model with custom loss function

### ISSUE THREAD

**Issue by khsjursen** [ODINN-SciML/MassBalanceMachine#26]
Model training with custom loss function gives warning that parameter `metadata_shape` is not being used:

[22:43:47] WARNING: /home/conda/feedstock_root/build_artifacts/xgboost-split_1667849527992/work/src/learner.cc:767:
Parameters: { "metadata_shape" } are not used. 

Warning is provided each time we run training on a fold. 
In `CustomXGBoostRegressor` class `metadata_shape` should be used to specify the number of rows in the feature matrix that are considered metadata and are not to be used in training.

```
    def __init__(self, metadata_shape, **kwargs):
        super(CustomXGBoostRegressor, self).__init__(**kwargs)
        self.metadata_shape = metadata_shape
```

**Comment by JulianBiesheuvel** [comment]
Fixed!
